api/services/EmailService.js

```diff
@@ -28,21 +30,27 @@ const EmailService = {
             user_ref: user_ref,
             service_type: serviceType,
         };
-        EmailService._sendRequestToNotificationService(postData, url);
+
+        await EmailService._sendRequestToNotificationService(postData, url);
     },
-    failedDocuments(email, failed_certs) {
+    async failedDocuments(email, failed_certs) {
         const url = '/failed-documents';
         const postData = { to: email, failed_certs: failed_certs };
-        EmailService._sendRequestToNotificationService(postData, url);
+
+        await EmailService._sendRequestToNotificationService(postData, url);
     },
 
-    _sendRequestToNotificationService(postData, url) {
-        emailRequest.post(url,{ data: postData }).then((response) => {
-            sails.log.info(response.status, response.data);
-        }).catch((err) => {
+    async _sendRequestToNotificationService(postData, url) {
+        try {
+            const res = await emailRequest.post(url, postData);
```

Comment:
This is the main thing that needed to be changed. The rest of the changes are more for unit tests.